File: docs-examples/example-python/src/test/python/micronaut/configuration/openapi/docs/SwaggerAnnotationsTest.py
from micronaut.test.extensions.junit5.annotation import MicronautTest
from org.junit.jupiter.api import Test

from . import OpenApiSpec


@MicronautTest
class SwaggerAnnotationsTest:

    @Test
    def test_swagger_annotations_take_precedence(self):
        operation = OpenApiSpec.load().getPaths().get("/greetings/{name}").getGet()

        assert list(operation.getTags()) == ["greeting"]
        assert operation.getSummary() == "Greets a person"
        assert operation.getDescription() == "A friendly greeting is returned"
        assert operation.getOperationId() == "greetings"
        parameter = operation.getParameters().get(0)
        assert parameter.getName() == "name"
        assert parameter.getIn() == "path"
        assert parameter.getDescription() == "The name of the person"
        assert parameter.getRequired()
        assert parameter.getSchema().getMinLength() == 1
        assert parameter.getSchema().getType() == "string"
        responses = operation.getResponses()
        assert responses.get("200").getContent().get("text/plain").getSchema().getType() == "string"
        assert responses.get("400").getDescription() == "Invalid Name Supplied"
        assert responses.get("404").getDescription() == "Person not found"

File: docs-examples/example-java/src/test/java/io/micronaut/configuration/openapi/docs/SwaggerAnnotationsTest.java
package io.micronaut.configuration.openapi.docs;

import org.junit.jupiter.api.Test;

import java.util.List;

import static org.junit.jupiter.api.Assertions.assertEquals;
import static org.junit.jupiter.api.Assertions.assertTrue;

class SwaggerAnnotationsTest {

    @Test
    void swaggerAnnotationsTakePrecedence() {
        var operation = OpenApiSpec.load().getPaths().get("/greetings/{name}").getGet();

        assertEquals(List.of("greeting"), operation.getTags());
        assertEquals("Greets a person", operation.getSummary());
        assertEquals("A friendly greeting is returned", operation.getDescription());
        assertEquals("greetings", operation.getOperationId());
        var parameter = operation.getParameters().get(0);
        assertEquals("name", parameter.getName());
        assertEquals("path", parameter.getIn());
        assertEquals("The name of the person", parameter.getDescription());
        assertTrue(parameter.getRequired());
        assertEquals(1, parameter.getSchema().getMinLength());
        assertEquals("string", parameter.getSchema().getType());
        var responses = operation.getResponses();
        assertEquals("string", responses.get("200").getContent().get("text/plain").getSchema().getType());
        assertEquals("Invalid Name Supplied", responses.get("400").getDescription());
        assertEquals("Person not found", responses.get("404").getDescription());
    }
}
